src/shared_modules/indexer_connector/src/indexerConnector.cpp

```diff
@@ -383,32 +383,44 @@ IndexerConnector::IndexerConnector(
                 dataQueue.pop();
                 auto parsedData = nlohmann::json::parse(data);
                 const auto& id = parsedData.at("id").get_ref<const std::string&>();
+                // If the element should not be indexed, only delete it from the sync database.
+                const bool shouldIndex = !parsedData.contains("no-index") || !parsedData.at("no-index").get<bool>();
```

Comment:
Nitpick: We use a different logic in another file
```suggestion
                const bool shouldIndex = parsedData.contains("no-index") ? parsedData.at("no-index").get<bool>() : false;
```